### PR TITLE
[WIP] Teardown imrovement:

### PR DESCRIPTION
**Pull request by RazTamir**
Adding a verification in teardown for the PG state as it have to be clean_active.
 This is a generic validation for situations like in:
 https://github.com/red-hat-storage/ocs-ci/issues/1890

Signed-off-by: ratamir <ratamir@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -611,6 +611,23 @@ def get_rebalance_status(self):
                 and states['count'] == total_pg_count
             )
 
+    @retry(UnexpectedBehaviour, tries=20, delay=10, backoff=1)
+    def get_rebalance_status(self):
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:615]: change func name to something like "wait_for_rebalance_to_complete"

Comment (RazTamir) [reply ocs_ci/ocs/cluster.py:615]: fixed

**Comment by RazTamir** [comment]
The failures are exactly what this PR is all about.
Few examples:

1) pass teardown

16:03:55 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig rsh rook-ceph-tools-74494c7b4b-7fnnf ceph status -f json-pretty
16:03:56 - MainThread - ocs_ci.ocs.cluster - INFO - HEALTH_OK
16:03:56 - MainThread - ocs_ci.ocs.cluster - INFO - [{'state_name': 'active+clean', 'count': 136}]
16:03:56 - MainThread - tests.conftest - INFO - Ceph rebalance check passed at teardown
16:03:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
16:03:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
16:03:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-74494c7b4b-7fnnf ceph health
16:03:58 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
16:03:58 - MainThread - tests.conftest - INFO - Ceph health check passed at teardown


2) failed teardown

18:54:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig rsh rook-ceph-tools-74494c7b4b-7fnnf ceph status -f json-pretty
18:54:20 - MainThread - ocs_ci.ocs.cluster - INFO - HEALTH_OK
18:54:20 - MainThread - ocs_ci.ocs.cluster - INFO - [{'state_name': 'active+remapped+backfill_wait', 'count': 109}, {'state_name': 'active+clean', 'count': 40}, {'state_name': 'active+remapped+backfilling', 'count': 3}]
18:54:20 - MainThread - tests.conftest - INFO - Ceph rebalance check failed at teardown
18:54:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig rsh rook-ceph-tools-74494c7b4b-7fnnf ceph status -f json-pretty
18:54:21 - MainThread - ocs_ci.ocs.cluster - INFO - HEALTH_OK
18:54:21 - MainThread - ocs_ci.ocs.cluster - INFO - [{'state_name': 'active+remapped+backfill_wait', 'count': 109}, {'state_name': 'active+clean', 'count': 41}, {'state_name': 'active+remapped+backfilling', 'count': 2}]

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -590,25 +590,45 @@ def get_throughput_percentage(self):
         logging.info(f"The throughput percentage of the cluster is {throughput_percentage}%")
         return throughput_percentage
 
-    def get_rebalance_status(self):
+    def get_rebalance_status_base(self):
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:593]: if you are uploading one more commit, I suggest reverting this method name back as 'get_rebalance_status' name is not in use now

**Comment by RazTamir** [comment]
Executed once more:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/7062/consoleFull

Look for "11:43:57 - MainThread - tests.conftest - INFO - Ceph rebalance check failed at teardown"

**Comment by ebenahar** [comment]
As discussed with @RazTamir, we are going to wait for 4.4 GA before merging this PR as it can cause regressions

**Comment by RazTamir** [comment]
> As discussed with @RazTamir, we are going to wait for 4.4 GA before merging this PR as it can cause regressions.

----> intended regressions to improve quality :)
> 
> Note: we should send an email to OCS-QE after merging to notify about the upcoming automation failures that could be caused by this PR

**Comment by ebenahar** [comment]
@RazTamir  is this PR still relevant?

**Comment by petr-balogh** [comment]
Closing due to inactivity! Feel free to re-open or send the new PR.

**Pull request closed without merging**